sys/syz-extract/openbsd.go

```diff
@@ -57,21 +57,26 @@ func (*openbsd) processFile(arch *Arch, info *compiler.ConstInfo) (map[string]ui
 			args = append(args, "-I"+dir)
 		}
 	}
-	// Syscall consts on openbsd have weird prefixes sometimes,
-	// try to extract consts with these prefixes as well.
+	// Some syscalls on OpenBSD are prefixed with `SYS___' as opposed of the
+	// more common `SYS_' prefix.
+	syscallsQuirks := map[string]bool{
+		"SYS_get_tcb":      true,
+		"SYS_getcwd":       true,
+		"SYS_realpath":     true,
+		"SYS_semctl":       true,
+		"SYS_set_tcb":      true,
+		"SYS_syscall":      true,
+		"SYS_tfork":        true,
+		"SYS_threxit":      true,
+		"SYS_thrsigdivert": true,
+		"SYS_thrsleep":     true,
+		"SYS_thrwakeup":    true,
+		"SYS_tmpfd":        true,
+	}
 	compatNames := make(map[string][]string)
 	for _, val := range info.Consts {
-		const SYS = "SYS_"
-		if strings.HasPrefix(val, SYS) {
-			for _, prefix := range []string{"_", "__", "___"} {
-				for _, suffix := range []string{"30", "50"} {
-					compat := SYS + prefix + val[len(SYS):] + suffix
-					compatNames[val] = append(compatNames[val], compat)
-					info.Consts = append(info.Consts, compat)
-				}
-			}
-		} else {
-			compat := "LINUX_" + val
```

Comment:
The changes look fine to me in general, but I don't understand the OpenBSD specifics/details. I don't mind merging this, but maybe @blackgnezdo is a better reviewer.

This "LINUX_" and "30"/"50" thing above, don't we lose something here?
On a related note: I assume you re-run this new version on new OpenBSD. Are there any changes? It's probably a good idea to submit new .const files as well.

Reply:
> The changes look fine to me in general, but I don't understand the OpenBSD specifics/details. I don't mind merging this, but maybe @blackgnezdo is a better reviewer.
> 
> This "LINUX_" and "30"/"50" thing above, don't we lose something here?
> On a related note: I assume you re-run this new version on new OpenBSD. Are there any changes? It's probably a good idea to submit new .const files as well.

Those compat defines are not defined on OpenBSD anyway. The only syscall in with a `SYS___` prefix is semctl. I just took the opportunity to complete the full table.

No changes to const files after this change.